rio/src/RIO/Prelude/Logger.hs

```diff
@@ -8,7 +8,10 @@ module RIO.Prelude.Logger
   , logError
   , logOther
     -- * Running with logging
+  , newLogFunc
```

Comment:
It doesn't seem strictly relevant to add in a `new` variant of these functions in addition to the `with`, unless I'm missing something.

Reply:
I guess it depends on the use case; I'm using [other means ](https://github.com/roman/Haskell-componentm) to manage the lifecycle of the resources allocated by my application. I think this won't be much of a need if I can reload the `LogFunc` behavior after it is built.

Reply:
I'd rather consider the addition of these functions separately then, and ideally figure out a way to avoid exposing them.

Reply:
I'll move this out of this PR and create another one with those changes only